clang/lib/Driver/Driver.cpp

```diff
@@ -4995,14 +4995,33 @@ void Driver::BuildActions(Compilation &C, DerivedArgList &Args,
 
   // For an FPGA archive, we add the unbundling step above to take care of
   // the device side, but also unbundle here to extract the host side
-  for (const auto &LI : LinkerInputs) {
+  for (auto &LI : LinkerInputs) {
     Action *UnbundlerInput = nullptr;
+    auto wrapObject = [&]() {
+      bool EarlyLink = false;
+      if (const Arg *A = Args.getLastArg(options::OPT_fsycl_link_EQ))
+        EarlyLink = A->getValue() == StringRef("early");
+      if (Args.hasArg(options::OPT_fintelfpga) && EarlyLink) {
```

Comment:
Quicker to check?
```suggestion
      if (EarlyLink && Args.hasArg(options::OPT_fintelfpga)) {
```

Reply:
Yup, should be quicker - done.

Reply:
Being a newbie, can you please explain the difference between the two? Thanks :)!

Reply:
@meiyacha, the difference is the order in which the values within the conditional are verified.  As this is an `&&` condition, both values have to be `true`.  By evaluating the boolean first, we can skip the condition if false right away and not have to evaluate the `Args.hasArg` check which is more labor intensive as we are parsing the entire argument list.